RoadTripMoustache/kana_to_kanji
# feat(feedback): Implement feedback feature! (#28)

## 📖 Description


Implementation of feedback option. This option can be found under the _General_ section of the settings.
It will open a bottom sheet with the option to create a **feature request** or **report a bug**. It will then ask the user to give more information on the feedback he/she wants to offer, and finally, create a GitHub issue on this repository

The user also has the option to take a screenshot of the application using the [Feedback](https://pub.dev/packages/feedback).

#### Rules applied on the feedback form

- **email**: optional on both cases, when the user profile is added, it will become a checkbox 'send as Anonyme'
- **description**: only required for `feature request`
- **steps to reproduce**: only required for `bug report`
- **screenshot**: only available for `bug report` but optional.

#### Other changes

- Introductions of `DialogService`, a wrapper allowing use to create dialogs without the context
- Enhancement of `InfoService` to have device information like the device model
- Adding a simple extension to get a `MaterialColor` from a `Color`

### ⁉️ Related Issues



closes #28 


### 🖼️ Screenshots:


<details>
    <summary>Demonstration 'Feature request' on iOS, the issue created --> #89 </summary> 

https://github.com/RoadTripMoustache/kana_to_kanji/assets/22211097/84b213e6-73cd-4b7a-a44f-cb154f161604

</details>

<details>
    <summary>Demonstration 'Report bug' on iOS, the issue created --> #90 </summary> 

https://github.com/RoadTripMoustache/kana_to_kanji/assets/22211097/1aa402c1-6199-4b12-a7c2-27b0e95aa9ec

</details>

The Android build looks the same, except for one detail. On the screenshot screen, the _Report_ button is more prominent, I try to find a solution without success.

### 🧪 How to test the change?




1. If you are testing locally, make sure to update your `.env`, run `flutter clean` and regenerate the files
2. Go to the _Settings_ view
3. Click on _Give feedback_ button
4. See changes

### ☑️ Checklist before requesting a review
